cordapp/README.md

```diff
@@ -95,7 +107,337 @@ GqO4M1KlfimphQwCICiq00hDanT5W8bTLqE7GIGuplf/O8AABlpWrUg6uiUB
 -----END CERTIFICATE-----
 ```
 
-### Note on conclave modes
-By default, this sample will build and run in [mock mode](https://docs.conclave.net/mockmode.html), and so won't use a
-secure enclave. For a list of modes and their properties, see [here](https://docs.conclave.net/enclave-modes.html).
-For instructions on how to set the mode at build time, see [here](https://docs.conclave.net/running-hello-world.html#beyond-mock-mode)
+### Note on Conclave modes
+By default, this sample will build and run in [mock mode](https://docs.conclave.net/mockmode.html), so it won't use a
+secure enclave. For a list of Conclave modes and their properties, see [here](https://docs.conclave.net/enclave-modes.html).
+For instructions on how to set the mode at build time, see [here](https://docs.conclave.net/running-hello-world.html#beyond-mock-mode).
+
+## Configure your workflow CorDapp module
+
+Both Conclave and Corda rely on Gradle build system plugins. Follow the instructions in the
+[hello world tutorial](https://docs.conclave.net/writing-hello-world.html) to configure Gradle to include an enclave mode. Add the host libraries
+to your Corda workflows module:
+
+```
+dependencies {
+    compile project(path: ":enclave", configuration: mode)
+    compile "com.r3.conclave:conclave-host:$conclaveVersion"
+    compile "com.r3.conclave:conclave-client:$conclaveVersion"
+    compile "com.r3.conclave:conclave-common:$conclaveVersion"
+    compile "com.r3.conclave:conclave-mail:$conclaveVersion"
+
+    // Corda dependencies.
+    cordaCompile "$corda_release_group:corda-core:$corda_release_version"
+    cordaRuntime "$corda_release_group:corda:$corda_release_version"
+    testCompile "$corda_release_group:corda-node-driver:$corda_release_version"
+}
+```
+
+*Note:
+You must use Gradle's `compile` configurations for the host dependencies. If you use `implementation`, you
+will get errors when the node starts up about missing classes due to issues with fat JARing. The host `dependencies` section should look like the above snippet.*
+
+## Write an enclave host service
+
+Conclave loads the enclave into a service object. This singleton will be available to flows for later usage, and lets
+us integrate with the node's lifecycle. The sample project contains a helper class called `EnclaveHostService` that
+you can copy into your project and subclass. You can edit this helper class to make it work for your use case.
+
+```java
+@CordaService
+public class ReverseEnclaveService extends EnclaveHostService {
+    public ReverseEnclaveService(@NotNull AppServiceHub serviceHub) {
+        super("com.r3.conclave.cordapp.sample.enclave.ReverseEnclave");
+    }
+}
+```
+
+This will enable SGX support on the host. Then, it loads the sample `ReverseEnclave` class. The `EnclaveHostService` class exposes methods to
```

Comment:
Hi @shamsasari, the expression "This will make a best effort" appears defensive and unsure on our part. That's the reason I changed it. Please let me know your opinion. Shall we change it to "This enables SGX support on the host if it's not enabled already."?

Reply:
Actually, the code doesn't do that. Perhaps it was removed as some point. You can remove that first sentence.

Reply:
Removed the sentence.